### PR TITLE
script-directory: add module

### PR DESCRIPTION
**Pull request by Janik-Haag**
### Description

Added a module for github.com/ianthehenry/sd
The [PR](https://github.com/NixOS/nixpkgs/pull/232287) was just merged yesterday https://nixpk.gs/pr-tracker.html?pr=232287

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (ncfavier) [inline .github/CODEOWNERS:1]: This file was removed recently

File: modules/programs/script-directory.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      type = lib.types.attrsOf lib.types.str;
+      example = lib.literalExpression ''
+        {
+          SD_ROOT = "''${config.home.homeDirectory}/.sd";
+          SD_EDITOR = "nvim";
+          SD_CAT = "lolcat";
+        }
+      '';
+      description = lib.mdDoc
+        "scirpt-directory config, for options take a look at the [documentation](github.com/ianthehenry/sd#options)";
```
Comment (ncfavier) [inline modules/programs/script-directory.nix:22]:
```suggestion
        "script-directory config, for options take a look at the [documentation](github.com/ianthehenry/sd#options)";
```

**Comment by Janik-Haag** [comment]
I don't get why the macos tests are failing, the linux ones are building fine. I don't have darwin machine, so I sadly can't test what actually breaks but I can't really imagine it being related to this PR because everything in here should work fine.

**Comment by ncfavier** [comment]
The macos tests should be fixed on master now.

**Comment by Janik-Haag** [comment]
Can you trigger the macos test runner for this pr again? And I assume you want to wait with merging until script-directory reached unstable?

**Comment by ncfavier** [comment]
You'll have to rebase on master, and yes.

**Comment by Janik-Haag** [comment]
@ncfavier script-directory is on unstable now

**Comment by ncfavier** [comment]
Could you add a [news entry](https://nix-community.github.io/home-manager/index.html#sec-guidelines-news-style) please?

**Comment by Janik-Haag** [comment]
Added a news entry

File: modules/misc/news.nix [hunk 2]
```diff
@@ -1033,6 +1033,12 @@ in
           defaults to 'pkgs.openssh'.
         '';
       }
+      {
+        time = "2023-05-18T21:03:30+00:00";
+        message = ''
+          A new module is available: 'programs.shell-directory'.
```
Comment (dawidd6) [inline modules/misc/news.nix:1039]: Shouldn't this be `programs.script-directory`?

Comment (ncfavier) [reply modules/misc/news.nix:1039]: Thanks, fixed

**Pull request merged**